views/login.go

```diff
@@ -76,23 +52,17 @@ func (v *Views) LoginFunc(w http.ResponseWriter, r *http.Request) {
 			MsgType: "",
 		}
 
-		err = v.template.RenderNoNavsTemplate(w, data, templates.LoginTemplate)
-		//err = v.tpl.ExecuteTemplate(w, "login", context)
-		if err != nil {
-			log.Printf("login failed to exec tmpl: %+v", err)
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-		}
-		return
+		return v.template.RenderNoNavsTemplate(c.Response(), data, templates.LoginTemplate)
 	case "POST":
 		// Parsing form to struct
-		err = r.ParseForm()
+		err = c.Request().ParseForm()
 		if err != nil {
 			log.Printf("parse form fail: %v", err)
-			return
+			return v.errorHandle(c, err)
 		}
-		username := r.FormValue("username")
-		password := r.FormValue("password")
-		u := user.User{}
+		username := c.Request().FormValue("username")
+		password := c.Request().FormValue("password")
```

Comment:
`c.FormValue()` is shorter and more idiomatic, or consider [binding](https://echo.labstack.com/docs/binding)

Reply:
I didn't realise that existed, I'll replace them all